### ISSUE TITLE
Schedule An Import fails via Wordpress importer plugin

### ISSUE THREAD

**Issue by SnakeME** [matomo-org/plugin-GoogleAnalyticsImporter#112]
We have completed the steps creating an OAuth Client and authorizing Matomo to import data from Google Analytics.

On the "Schedule An Import" step we've filled out the fields Google Analytics Property ID, Google Analytics Account ID, Google Analytics View ID. When we then click on "Start" we receive the following error:

The following error just broke Matomo (v3.14.0):

Neither the property "hasNoClear" nor one of the methods "hasNoClear()", "gethasNoClear()"/"ishasNoClear()" or "__call()" exist and have public access in class "__PHP_Incomplete_Class".
in /usr/www/users/abcdefg/wp-content/plugins/matomo/app/plugins/CoreHome/templates/_notifications.twig line 5

**Comment by tsteur** [comment]
@SnakeME this sounds like some incompatibility issue with another plugin. By the sounds another plugin maybe handles session data in a different way and breaks some of our functionality. Could you post the output of the system report here as a comment (the data should be anonymised. see https://matomo.org/faq/wordpress/how-do-i-fix-the-error-your-php-installation-appears-to-be-missing-the-mysql-extension-which-is-required-by-wordpress-in-matomo-system-report/ ) or send it to us by email to wordpress at matomo.org ? If you choose email by great to comment afterwards so we can check the spam folder if needed.

**Comment by SnakeME** [comment]
Hi @tsteur, it's strange, because now after trying a couple of days later, I get the following error:

Das Datumsformat muss YYYY-MM-DD, or 'today' or 'yesterday' oder eine andere, von der Funktion strtotime unterstützte Zeichenkette sein.

This are the headers which were sent for the request:

module: GoogleAnalyticsImporter
action: startImport
startDate: 2013-06-01
endDate: 2020-08-28
propertyId: UA-XXXXXXX-1
viewId: 73207725
nonce: 7bcb9e9be214da44cb2c9d8989f1163b
accountId: XXXXXXXXXX
isMobileApp: 0
timezone: 
extraCustomDimensions[0][gaDimension]: 
extraCustomDimensions[0][dimensionScope]: 
extraCustomDimensions[0][$$hashKey]: object:57
isVerboseLoggingEnabled: 0
format: JSON2
idSite: 1
period: day
date: yesterday

Maybe some problem with the PHP version or another incompatibility? We are running PHP 7.4 on a dedicated server.

**Comment by tsteur** [comment]
I think @diosmosis fixed this particular error just before in https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/pull/117/files. If you can it be great to still get your system report then we could identify the root cause maybe for the first error.

**Comment by SnakeME** [comment]
Here is the system report

# Matomo

* Matomo Pluginversion: 1.2.0
* Config existiert und ist beschreibbar.: Ja ("$ABSPATH/wp-content/uploads/matomo/config/config.ini.php" )
* JS Tracker existiert und ist beschreibbar.: Ja ("$ABSPATH/wp-content/uploads/matomo/matomo.js" )
* Plugin-Verzeichnisse: Yes ([{"pluginsPathAbsolute":"$ABSPATH\/wp-content\/plugins\/matomo\/plugins","webrootDirRelativeToMatomo":"..\/"},{"pluginsPathAbsolute":"$ABSPATH\/wp-content\/plugins","webrootDirRelativeToMatomo":"..\/..\/"}])
* Tmp-Verzeichnis schreibbar: Yes ($ABSPATH/wp-content/cache/matomo)
* Matomo-Version: 3.14.0
* Matomo Blog idSite: 1
* Installierte Matomo-Version: 1.1.3 (Install date: 2020-07-05 08:38:05)

## Endpoints

* Matomo JavaScript Tracker URL:  ($site_url/wp-content/uploads/matomo/matomo.js)
* Matomo JavaScript Tracker - WP Rest API:  ($site_url/wp-json/matomo/v1/hit/)
* Matomo HTTP Tracking API:  ($site_url/wp-content/plugins/matomo/app/matomo.php)
* Matomo HTTP Tracking API - WP Rest API:  ($site_url/wp-json/matomo/v1/hit/)

## Crons

* Server-Zeit: 2020-08-28 06:32:18
* Blog-Zeit: 2020-08-28 06:32:18 (Die nachfolgenden Daten werden in der Blog-Zeitzone angezeigt)
* Sync users & sites: Next run: 2020-08-28 08:38:04 (2 Stunden 5 Minuten) ( Last started: 2020-08-27 08:38:05 (-21 Stunden 54 Minuten). Last ended: 2020-08-27 08:38:05 (-21 Stunden 54 Minuten). Interval: daily)
* Archive: Next run: 2020-08-28 06:38:04 (5 Minuten 46s) ( Last started: 2020-08-28 05:38:34 (-53 Minuten 44s). Last ended: 2020-08-28 05:38:38 (-53 Minuten 40s). Interval: hourly)
* Update GeoIP DB: Next run: 2020-09-03 08:38:04 (6 Tage 2 Stunden) ( Last started: 2020-08-04 08:38:07 (-23 Tage 21 Stunden). Last ended: 2020-08-04 08:38:08 (-23 Tage 21 Stunden). Interval: matomo_monthly)

## Vorgeschriebene Prüfungen

* PHP-Version >= : ok 
* PDO Erweiterung: ok 
* PDO\MYSQL Erweiterung: ok 
* MYSQLI Erweiterung: ok 
* Weitere erforderliche Anforderungen: ok 
* Erforderliche Funktionen: ok 
* Benötigte PHP Konfiguration (php.ini): ok 
* Verzeichnisse mit Schreibzugriff: ok 
* Verzeichnisse mit Schreibzugriff auf Tag Manager: ok 

## Optionale Prüfungen

* Tracker-Status: ok 
* Speicherlimit: ok 
* Zeitzone: ok 
* Öffnen einer URL: ok 
* PageSpeed deaktiviert: ok 
* GD > 2.x + Freetype (graphics): ok 
* Andere Erweiterungen: ok 
* Andere Funktionen: ok 
* Dateisystem: ok 
* Archiv Cron: ok 
* Letzter erfolgreicher Abschluss der Archivierung: ok 
* Maximale Packetgröße: ok 
* Standorterkennung: ok 
* Update über HTTPS: ok 
* Schreibbarer JavaScript-Tracker ("/matomo.js"): ok 
* [GoogleAnalyticsImporter] Required PHP Functions: ok 
* [GoogleAnalyticsImporter] Required Executables (php): ok 
* [GoogleAnalyticsImporter] Required Executables (nohup): ok 
* Supports Async Archiving: No
* Location provider ID: geoip2php
* Location provider available: Yes
* Location provider working: Yes
* Had visit in last 5 days: Yes

## Matomo Settings

* Track mode: default
* Track codeposition: footer
* Track api endpoint: default
* Track js endpoint: default
* Version history: 1.2.0, 1.1.3
* Core version: 3.14.0
* Last tracking settings update: 1598347801
* Last settings update: 1598347801
* Show get started page: 0
* Track ecommerce: No
* Disable cookies: Yes
* Delete all data uninstall: No

## Logs

* Keine: 

# WordPress

* Home URL: $site_url
* Site URL: $site_url
* WordPress Version: 5.4.2
* Number of blogs: 1
* Multisite Enabled: No
* Network Enabled: No
* WP_DEBUG: No
* WP_DEBUG_DISPLAY: Yes
* WP_DEBUG_LOG: No
* DISABLE_WP_CRON: -
* FORCE_SSL_ADMIN: Yes
* WP_CACHE: Yes
* CONCATENATE_SCRIPTS: -
* COMPRESS_SCRIPTS: -
* COMPRESS_CSS: -
* ENFORCE_GZIP: -
* WP_LOCAL_DEV: -
* DIEONDBERROR: -
* WPLANG: de_DE
* ALTERNATE_WP_CRON: -
* WP_CRON_LOCK_TIMEOUT: 60
* WP_DISABLE_FATAL_ERROR_HANDLER: -
* Permalink Structure: /%postname%/
* Possibly uses symlink: No
* Compatible content directory: Yes

# WordPress Plugins

* A-Z Listing: 3.1.0
* Advanced WP Columns: 2.0.6
* Ajax Search Lite: 4.8.3
* Broken Link Checker: 1.11.13
* Classic Editor: 1.5
* Contact Form 7: 5.2.1
* Cookie Notice: 1.3.2
* Custom Permalinks: 1.5.1
* Display Posts: 3.0.2
* Duplicate Post: 3.2.4
* Fast Secure Contact Form: 3.1.8.3
* Google Analytics Dashboard for WP (GADWP): 6.1.0
* Google Analytics Importer (Matomo Plugin): 1.5.1
* Insert HTML Snippet: 1.3.1
* Matomo Analytics - Ethical Stats. Powerful Insights.: 1.2.0
* Matomo Marketplace for WordPress: 1.0.5
* Mini Loops: 1.3.1
* Page-list: 5.2
* Redirection: 4.8
* Relevanssi: 4.7.2.1
* SEO Smart Links: 3.0.1
* Shortcode Exec PHP: 1.46
* SI CAPTCHA Anti-Spam: 2.7.6.4
* UpdraftPlus - Backup/Restore: 1.16.26
* W3 Total Cache: 0.14.2 (Network enabled)
* Widget Context: 1.3.2
* WP Dropdown: 0.1
* WP Permalauts: 1.0.2
* Yoast SEO: 14.4.1
* Active Plugins: 26 (GoogleAnalyticsImporter a-z-listing advanced-wp-columns ajax-search-lite broken-link-checker contact-form-7 cookie-notice custom-permalinks display-posts-shortcode duplicate-post google-analytics-dashboard-for-wp insert-html-snippet matomo-marketplace-for-wordpress matomo mini-loops page-list redirection relevanssi seo-automatic-links shortcode-exec-php updraftplus w3-total-cache widget-context wordpress-seo wp-dropdown wp-permalauts)

# Server

* Server Info: Apache
* PHP OS: Linux
* PHP Version: 7.4.9
* PHP SAPI: cgi-fcgi
* PHP Binary Name: php74.bin.fcgi
* PHP Found Binary: php -q
* Timezone: UTC
* WP timezone: +00:00
* Locale: de_DE
* User Locale: de_DE
* Memory Limit: 256M (At least 128MB recommended. Depending on your traffic 256MB or more may be needed.)
* WP Memory Limit: 40M
* WP Max Memory Limit: 256M
* Time: 1598596338
* Max Execution Time: 30
* Max Post Size: 5M
* Max Upload Size: 5242880
* Max Input Vars: 1000
* Disabled PHP functions: No
* zlib.output_compression is off: Yes
* Curl Version: 7.64.0, OpenSSL/1.1.1d
* Suhosin installed: No

# Database

* MySQL Version: 5.7.31
* Mysqli Connect: Yes
* Force MySQL over Mysqli: No
* DB Prefix: kas_wp_
* DB CHARSET: utf8
* DB COLLATE: 
* SHOW ERRORS: No
* SUPPRESS ERRORS: No
* Uses Socket: No
* Uses IPv6: No
* Matomo tables found: 79
* Matomo users found: 2
* Matomo sites found: 1
* Erforderliche Berechtigungen: OK

# Browser

* Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.135 Safari/537.36
* Language: de-de,de,en-us,en

**Comment by SnakeME** [comment]
> I think @diosmosis fixed this particular error just before in https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/pull/117/files. If you can it be great to still get your system report then we could identify the root cause maybe for the first error.

I have updated the file and the Import Job was now created successfully. I don't know what the origin problem was regarding the "hasNoClear" error. Maybe the issue relates to the Google API because I've created the credentials just before I tried to create the job. Maybe something on Google's side was not ready yet.

**Comment by tsteur** [comment]
Thanks @SnakeME 

I've installed the same plugins but couldn't reproduce the session issue. I'd ignore it for now and see if it happens again in the future for some other user and then investigate further. Let us know if it appears again otherwise.

fyi @diosmosis I saw some code in GA importer was also starting a session in a class named `Random` within the vendor directory but not sure if that class would have been actually used etc.

Let's see if this happens again.
